script/deploy/mainnet/M2_Mainnet_Upgrade.s.sol

```diff
@@ -0,0 +1,308 @@
+            IETHPOSDeposit(ETHPOSDepositAddress),
+            eigenPodBeacon,
+            strategyManager,
+            slasher,
+            delegationManager
+        );
+        delayedWithdrawalRouterImplementation = new DelayedWithdrawalRouter(eigenPodManager);
+    }
+
+    function _simulateUpgrade() internal {
+
+        vm.startPrank(executorMultisig);
+
+        // First, upgrade the proxy contracts to point to the implementations
+        // AVSDirectory
+        // eigenLayerProxyAdmin.upgrade(
+        //     TransparentUpgradeableProxy(payable(address(avsDirectory))),
+        //     address(avsDirectoryImplementation)
+        // );
+        // DelegationManager
+        eigenLayerProxyAdmin.upgrade(
+            TransparentUpgradeableProxy(payable(address(delegationManager))),
+            address(delegationManagerImplementation)
+        );
+        // StrategyManager
+        eigenLayerProxyAdmin.upgrade(
+            TransparentUpgradeableProxy(payable(address(strategyManager))),
+            address(strategyManagerImplementation)
+        );
+        // Slasher
+        eigenLayerProxyAdmin.upgrade(
+            TransparentUpgradeableProxy(payable(address(slasher))),
+            address(slasherImplementation)
+        );
+        // EigenPodManager
+        eigenLayerProxyAdmin.upgrade(
+            TransparentUpgradeableProxy(payable(address(eigenPodManager))),
+            address(eigenPodManagerImplementation)
+        );
+        // Delayed Withdrawal Router
+        eigenLayerProxyAdmin.upgrade(
+            TransparentUpgradeableProxy(payable(address(delayedWithdrawalRouter))),
+            address(delayedWithdrawalRouterImplementation)
+        );
+
+        // Second, configure additional settings and paused statuses
+        delegationManager.setMinWithdrawalDelayBlocks(DELEGATION_MANAGER_MIN_WITHDRAWAL_DELAY_BLOCKS);
+        delegationManager.unpause(0);
+        eigenPodManager.unpause(0);
+
+        eigenPodManager.setDenebForkTimestamp(EIGENPOD_MANAGER_DENEB_FORK_TIMESTAMP);
+        eigenPodManager.updateBeaconChainOracle(beaconOracle);
+        eigenPodBeacon.upgradeTo(address(eigenPodImplementation));
+
+        vm.stopPrank();
+    }
+}
+
+// forge t --mt test_queueUpgrade --fork-url $RPC_MAINNET -vvvv
+contract Queue_M2_Upgrade is M2_Mainnet_Upgrade, TimelockEncoding {
+    Vm cheats = Vm(HEVM_ADDRESS);
+
+    // Thurs Apr 04 2024 12:00:00 GMT-0700 (Pacific Daylight Time)
+    uint256 timelockEta = 1712214000;
+
+    function test_queueUpgrade() external {
+        _parseDeployedContracts("script/output/mainnet/M1_deployment_mainnet_2023_6_9.json");
+        _parseInitialDeploymentParams("script/configs/mainnet/M2_mainnet_upgrade.config.json");
+
+        // TODO: fill in correct addresses
+        // simulate deploying contracts
+        _deployImplementationContracts();
+        // confirmed correct address
+        address beaconChainOracle = 0x343907185b71aDF0eBa9567538314396aa985442;
```

Comment:
I don't think this is needed? The deployment parser picks this up from the JSON config

Reply:
good point, will resolve

Reply:
fixed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/480/commits/9b7a0c33c94c20f2e5bb8abfacf115e15bf24426

Reply:
thanks, i feel better about this :)